dvc/repo/artifacts.py

```diff
@@ -0,0 +1,44 @@
+import os
+from typing import TYPE_CHECKING, Any, Dict
+
+from dvc.annotations import Artifact
+from dvc.dvcfile import FileMixin
+from dvc.utils import relpath
+
+if TYPE_CHECKING:
+    from dvc.repo import Repo
+
+
+class ArtifactsFile(FileMixin):
+    from dvc.schema import ARTIFACTS_SCHEMA as SCHEMA
+
+    def dump(self, stage, **kwargs):
+        raise NotImplementedError
+
+    def merge(self, ancestor, other, allowed=None):
+        raise NotImplementedError
+
+
+class Artifacts:
+    repo: "Repo"
+
+    def __init__(self, repo) -> None:
```

Comment:
No need to set at class level, you can just do:
```suggestion
    def __init__(self, repo: "Repo") -> None:
```